packages/connector-saml/src/index.ts

```diff
@@ -0,0 +1,178 @@
+import type {
+  GetAuthorizationUri,
+  GetUserInfo,
+  SocialConnector,
+  CreateConnector,
+  GetConnectorConfig,
+  ValidateSamlAssertion,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  validateConfig,
+  ConnectorType,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import * as saml from 'samlify';
+import { z } from 'zod';
+
+import { defaultMetadata, signingAlgorithmMapping } from './constant.js';
+import { samlConfigGuard } from './types.js';
+import type { SamlConfig } from './types.js';
+import { samlAssertionHandler, getUserInfoFromRawUserProfile } from './utils.js';
+
+const getAuthorizationUri =
+  (getConfig: GetConnectorConfig): GetAuthorizationUri =>
+  async ({ state, redirectUri, connectorId, connectorFactoryId, jti }, setSession) => {
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig<SamlConfig>(config, samlConfigGuard);
+    const {
+      entityID,
+      x509Certificate,
+      idpMetadataXml,
+      signAuthnRequest,
+      privateKey,
+      privateKeyPass,
+      nameIDFormat,
+      requestSignatureAlgorithm,
+      assertionConsumerServiceUrl,
+    } = samlConfigGuard.parse(config);
+
+    assert(
+      connectorId,
+      new ConnectorError(ConnectorErrorCodes.InvalidRequestParameters, {
+        message: 'Can not find string-typed variable `connectorId` from connector session.',
+      })
+    );
+
+    assert(
+      connectorFactoryId,
+      new ConnectorError(ConnectorErrorCodes.InvalidRequestParameters, {
+        message: 'Can not find string-typed variable `connectorFactoryId` from connector session.',
+      })
+    );
+
+    assert(
+      setSession,
+      new ConnectorError(ConnectorErrorCodes.NotImplemented, {
+        message: 'Function `setSession()` is not implemented.',
+      })
+    );
+
+    const storage = { state, redirectUri, connectorId, connectorFactoryId, jti };
+    await setSession(storage);
+
+    try {
+      // eslint-disable-next-line new-cap
+      const identityProvider = saml.IdentityProvider({
+        wantAuthnRequestsSigned: signAuthnRequest,
+        metadata: idpMetadataXml,
+      });
```

Comment:
nit: Looks like all the inputs are static configs. Is it ok if we keep a shared instance of identityProvider or have to generate it on each request?

Reply:
We aligned this implementation with other connectors (such as the AAD connector). The `config` could be updated during each connector's method calls (although it should not happen), so we create a new instance everytime we use it.